src/util/OGCFilter.js

```diff
@@ -0,0 +1,328 @@
+                filters, 'wms', combinator);
+        },
+
+        /**
+         * Given an array of ExtJS Filters, this method will return an OGC
+         * compliant filter which can be used for WFS requests
+         * @param {array} filters array containing all `Ext.grid.filters.filter`
+         *   that should be converted
+         * @param {string} combinator The combinator used for combining multiple
+         *   filters. Can be 'and' or 'or'
+         * @param {string} wfsVersion The WFS version to use, either `1.0.0`,
+         *   `1.1.0` or `2.0.0`
+         * @return {string} The OGC Filter XML
+         */
+        getOGCWFSFilterFromExtJSFilter: function(filters, combinator,
+            wfsVersion) {
+            return GeoExt.util.OGCFilter.getOGCFilterFromExtJSFilter(
+                filters, 'wfs', combinator, wfsVersion);
+        },
+
+        /**
+         * Given an ExtJS Filter, this method will return an OGC compliant
+         * filter which can be used for WMS or WFS queries
+         * @param {array} filters array containing all `Ext.grid.filters.filter`
+         *   that should be converted
+         * @param {string} type The OGC type we will be using, can be
+         *   `wms` or `wfs`
+         * @param {string} combinator The combinator used for combining multiple
+         *   filters. Can be 'and' or 'or'
+         * @param {string} wfsVersion The WFS version to use, either `1.0.0`,
+         *   `1.1.0` or `2.0.0`
+         * @return {string} the OGC Filter as XML tring
+         */
+        getOGCFilterFromExtJSFilter: function(filters, type, combinator,
+            wfsVersion) {
+            if (!Ext.isDefined(filters) || !Ext.isArray(filters)) {
+                Ext.Logger.error('Invalid filter argument given to ' +
+                  'GeoExt.util.OGCFilter. You need to pass an array of ' +
+                  '"Ext.grid.filters.filter"');
+                return;
+            }
+
+            var omitNamespaces = false;
+            // filters for WMS layers need to omit the namespaces
+            if (!Ext.isEmpty(type) && type.toLowerCase() === 'wms') {
+                omitNamespaces = true;
+            }
+            var ogcFilters = [];
+            Ext.each(filters, function(filter) {
+                var property = filter.config.property;
+                var operator = filter.config.operator;
+                var value = filter.getValue();
+                if (filter.isDateValue) {
+                    if (filter.getDateFormat) {
+                        value = Ext.Date.format(
+                            filter.getValue(),
+                            filter.getDateFormat()
+                        );
+                    } else {
+                        value = Ext.Date.format(filter.getValue(), 'Y-m-d');
+                    }
+                }
+                if (Ext.isEmpty(property) || Ext.isEmpty(operator) ||
+                    Ext.isEmpty(value)) {
```

Comment:
Wouldn't it make sense that the check is done before processing the date format?

Reply:
right, fixed